### PR TITLE
Retrieve owning tenant's identifier in java client responses

### LINKED ISSUE #13564: Add experimental tenant ownership to command responses
All APIs, where tenant ownership can be inferred, must add to their response which tenant owns the resulting entity. This way, users are aware of this information.

### PR DESCRIPTION
**Pull request by korthout**
## Description

This is the fourth (and last) in several pull requests to:
- #13517

This pull request adds a new `getTenantId()` method to the command responses of the Java client. This allows users to understand which tenant owns the entity resulting from the command.

The method is only added to the responses of commands where the tenant can be explicitly set by the user or where the tenant is inferred from the access token. Those are:
- Create Process Instance
- Create Process Instance With Result
- Evaluate Decision
- Broadcast Signal
- Publish Message
- Activate Jobs (for each activated job individually)

In the commands where the tenant should already be clear, it is not provided. Specifically, those responses are typically completely empty anyways. The responses that are left unchanged are:
- Cancel Process Instance
- Modify Process Instance
- Delete Resource
- Complete Job
- Fail Job
- Update Job Retries
- Throw Error
- Resolve Incident
- Set Variables

I've considered Stream Jobs response out of scope, although those likely return the same response object as the response of the Activate Jobs command.

Again, please note that the new `getTenantId()` method does not really return a tenant identifier at this time. It only returns an empty string as dummy value. The feature is still under development. The API is added so others can already start using this method during development. I'll inform the relevant teams once this is available to them.

## Related issues

closes #13564 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by koevskinikola — APPROVED** [review]
Looks great @korthout! :rocket: 

I only left a small nitpick comment, but feel free to ignore it.

File: clients/java/src/main/java/io/camunda/zeebe/client/api/response/ProcessInstanceEvent.java [hunk 1]
```diff
@@ -28,4 +30,8 @@ public interface ProcessInstanceEvent {
 
   /** Unique key of the created process instance on the partition */
   long getProcessInstanceKey();
+
+  /** Tenant identifier that owns this process instance */
```
[the remark this answers is not in the document]
Comment (korthout) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/response/ProcessInstanceEvent.java:34]: I like the idea of cleaning up the documentation pattern applied, but I chose to continue the style that existed in each file.

There's an issue to actually [clean up the documentation style](https://github.com/camunda/zeebe/issues/9045) if you're interested in taking this on.

File: clients/java/src/main/java/io/camunda/zeebe/client/api/response/ProcessInstanceResult.java [hunk 2]
```diff
@@ -50,4 +51,8 @@ public interface ProcessInstanceResult {
    * @return de-serialized variables as the given type
    */
   <T> T getVariablesAsType(Class<T> variableType);
+
+  /** Tenant identifier that owns this process instance */
```
[the remark this answers is not in the document]
Comment (korthout) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/response/ProcessInstanceResult.java:55]: This file had a mix of styles, so I used what our [style guide recommends](https://github.com/camunda/zeebe/issues/9045)

**Comment by korthout** [comment]
bors merge

**Pull request merged**